### ISSUE TITLE
tree inference & data management

### ISSUE THREAD

**Issue by rvosa** [naturalis/barcode-constrained-phylogeny#23]
It is probably a good idea to run the steps incrementally, by which I mean: writing the alignments is a task that you can kill and then it resumes where it left of, until all families have an alignment. Same for the constraint trees: you probably want to have a tree for every alignment and store this in a folder structure. What you want to avoid is having to run everything in one giant loop that can never fail.

With that in mind, it's probably better to not have a temporary tree that has a throwaway name [here](https://github.com/naturalis/barcode-constrained-phylogeny/blob/main/src/create_tree.py#L66), but rather have a system where the raxml run assumes that there is a constraint tree for every alignment (and they probably have the same name but with different extensions).

**Comment by rvosa** [comment]
Think of it as a pipeline with different steps. You'd want to be able to do something like `make alignments` and it starts spitting out alignments, possibly in parallel. If you kill the job, it'll be able to pick up where you left off. If you do `make subtrees` it knows that it first needs to `make alignments` and `make constraints`, i.e. it needs to be aware of dependencies. This can be accomplished through a variety of workflow management tools, maybe the easiest one being SnakeMake.

**Comment by Naomivanes** [comment]
Changed the temporary constraint tree to one that gets saved in a folder. Will make a new issue to make a snakemake pipeline/rewrite the code to be able to run in a snakemake pipeline.
